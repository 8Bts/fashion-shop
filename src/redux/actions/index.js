import API from '../../api/index';

export const fetchAllItems = () => ({
  type: 'FETCH_ALL_ITEMS',
  payload: API.items.all(),
});

export const fetchItem = (id) => ({
  type: 'FETCH_ITEM',
  payload: API.items.get(id),
});

export const fetchAllCategories = () => ({
  type: 'FETCH_ALL_CATEGORIES',
  payload: API.categories.all(),
});

export const fetchCategory = (id) => ({
  type: 'FETCH_CATEGORY',
  payload: API.categories.get(id),
});

export const fetchAllUsers = () => ({
  type: 'FETCH_ALL_USERS',
  payload: API.categories.all(),
});

export const fetchUser = (id) => ({
  type: 'FETCH_USER',
  payload: API.categories.get(id),
});
