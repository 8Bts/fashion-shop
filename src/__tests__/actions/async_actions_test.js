import configureMockStore from 'redux-mock-store';
import promise from 'redux-promise-middleware';
import nock from 'nock';
import * as Action from '../../redux/actions/index';

const middlewares = [promise];
const mockStore = configureMockStore(middlewares);

describe('async actions', () => {
  let store;
  beforeEach(() => {
    store = mockStore({});
  });
  afterEach(() => {
    nock.cleanAll();
  });

  it('creates FETCH_ALL_ITEMS when fetching items has been done', () => {
    nock('https://fashion-shop-api.herokuapp.com')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
      })
      .get('/items')
      .reply(200, []);

    return store.dispatch(Action.fetchAllItems())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates FETCH_SINGLE_ITEM_FULFILLED when fetching single item has been done', () => {
    nock('https://fashion-shop-api.herokuapp.com')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
      })
      .get('/items/1')
      .reply(200, []);

    return store.dispatch(Action.fetchItem(1))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates FETCH_ALL_CATEGORIES_FULFILLED when fetching all categories has been done', () => {
    nock('https://fashion-shop-api.herokuapp.com')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
      })
      .get('/categories')
      .reply(200, []);

    return store.dispatch(Action.fetchAllCategories())
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('creates FETCH_CATEGORY_FULFILLED when fetching single category has been done', () => {
    nock('https://fashion-shop-api.herokuapp.com')
      .defaultReplyHeaders({
        'access-control-allow-origin': '*',
      })
      .get('/categories/1')
      .reply(200, []);

    return store.dispatch(Action.fetchCategory(1))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
