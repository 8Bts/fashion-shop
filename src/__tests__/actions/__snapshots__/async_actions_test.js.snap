// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions creates FETCH_ALL_CATEGORIES_FULFILLED when fetching all categories has been done 1`] = `
Array [
  Object {
    "type": "FETCH_ALL_CATEGORIES_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_ALL_CATEGORIES_FULFILLED",
  },
]
`;

exports[`async actions creates FETCH_ALL_ITEMS when fetching items has been done 1`] = `
Array [
  Object {
    "type": "FETCH_ALL_ITEMS_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_ALL_ITEMS_FULFILLED",
  },
]
`;

exports[`async actions creates FETCH_CATEGORY_FULFILLED when fetching single category has been done 1`] = `
Array [
  Object {
    "type": "FETCH_CATEGORY_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_CATEGORY_FULFILLED",
  },
]
`;

exports[`async actions creates FETCH_SINGLE_ITEM_FULFILLED when fetching single item has been done 1`] = `
Array [
  Object {
    "type": "FETCH_SINGLE_ITEM_PENDING",
  },
  Object {
    "payload": Array [],
    "type": "FETCH_SINGLE_ITEM_FULFILLED",
  },
]
`;
